dace/transformation/dataflow/otf_map_fusion.py

```diff
@@ -49,23 +53,48 @@ def can_be_applied(self, graph, expr_index, sdfg, permissive=False):
                 if prod != self.first_map_exit:
                     return False
 
-        # Equation Solvability: First map must induce a non-ambiguous set of writes for all possible reads of second map.
+        # Condition: First map fully covers second map
+        first_edge = graph.edges_between(self.first_map_exit, self.array)
+        if len(first_edge) > 1:
+            return False
```

Comment:
Are there any useful cases that will fail to match due to this check?

Reply:
I have never seen an SDFG where this has happened, so maybe remove?

Reply:
I am running out of time for this PR, so I simplified this for now

Reply:
the ConsolidateEdges simplification pass should make sure this is actually one, so we should be ok.